File: go/internal/replay/parserfixture/testdata/trees/ruby/order_service.rb
module Example
  class OrderService
    def initialize(repository)
      @repository = repository
    end

    def place(customer_id, skus)
      order = @repository.create(customer_id)
      skus.each { |sku| order.add_line(sku) }
      @repository.save(order)
    end

    def find(order_id)
      @repository.by_id(order_id)
    end
  end
end

File: go/internal/replay/parserfixture/testdata/trees/elixir/order_service.ex
defmodule Example.OrderService do
  @moduledoc "Coordinates order placement and lookup."

  alias Example.OrderRepository

  def place(customer_id, skus) do
    order = OrderRepository.create(customer_id)

    order =
      Enum.reduce(skus, order, fn sku, acc ->
        add_line(acc, sku)
      end)

    OrderRepository.save(order)
  end

  def find(order_id) do
    OrderRepository.by_id(order_id)
  end

  defp add_line(order, sku) do
    Map.update(order, :lines, [sku], &[sku | &1])
  end
end
